docs/design_guide.md

```diff
@@ -0,0 +1,285 @@
+
+Class for containing information about any given election, independent from an audit. The `Election` will be initialized using the results in whatever format is available (CVS w/ nos., CVR w/out nos., Tally).
+
+This class should mainly be used for extracting/using necessary information from the reported results of the election. A better design for this class can be made with more information about how reported election results are input to the system.
+
+**Methods** \*incomplete
+
+- `__init__()`
+
+
+### Contest
+
+Class to maintain necessary information about a specific contest within an election.
+
+
+**Notes:**
+- The `Election` and `Contest` classes should be linked in some way.
+  - Idea: each `Contest` has an attribute to hold the `Election` it is from, then an `Audit` can accept a `Contest` and can access its `Election` if necessary?
+
+
+### Audit
+
+An abstract class to define a *general* audit. It will define attributes and parameters common to all audits:
+
+- risk limit (alpha)
+- beta
+- maximum number of ballots to draw during audit
+- object containing information about election/contest to be audited
+  - could be `Election` with set of `Contest`s
+  - could be a single `Contest`
+- replacement
+
+*Note:* Some of the above will be provided by the users for each unique audit, but others might be defined by developers in the audit sub-classes and be constant values for each unique audit of that type.
+
+Example pseudocode:
+
+```python
+class audit:
+  alpha: float
+  beta: float
+  max_num_ballots: int
+  replacement: bool
+
+  def __init__(self, alpha, election, ...):
+    self.alpha = alpha
+    self.election = election
+    ...
+```
+
+
+**Methods** \**incomplete*
+
+- `__init__()`
+
+### Athena(Audit)
+
+A subclass of the audit class which defines the Athena audit.
+
+**Methods** \**incomplete*
+
+- `__init__()` (which also uses `super.__init__()`)
+- `stopping_prob()`
+  - takes previous roud sizes and kmins as input
+- Function to generate kmins
+  - will take in all previous round sizes and kmins
+- Function to advise next round size?
+
+
+
+### BRLA(Audit)
+
+A subclass of the audit class which defines the BRLA audit.
+
+**Method** \**incomplete*
+
+- `__init__()` (which also uses `super.__init__()`)
+- `kmins(round_schedule)`
+
+
+## Library File/Directory Hierarchy
+
+Based on [cookiecutter-pylibrary](https://github.com/ionelmc/cookiecutter-pylibrary) template
+
+For library code:
+
+```
+./src/r2b2/         # Directory to contain library source code
+    ./tests/        # Directory to contain tests within library source code
+    __init__.py     
+    __main__.py
+    cli.py          # In case we ever want to implement a CLI
+    ...
+    election.py     # Election class definition
+    contest.py      # Contest class definition
+    audit.py        # Abstract audit class definition  
+    bravo.py        # BARVO implemented as subclass of audit
+    ...
+```
+
+
+# Proposed Code Development Plan
+
+
+## Style Guide
```

Comment:
I think it would help to reference the source for the guidelines that we've adopted from elsewhere, both for attribution, and to help keep things coherent.

Reply:
Do you mean in addition to the link to PEP8?

Reply:
That was my thought. If a lot of this text comes straight from PEP8, it would help to note that and note things that are left out. Or if it comes from elsewhere, note those other places.
If you've got the refs at hand, please add them. If not, I don't want to hold things up so we can move on.